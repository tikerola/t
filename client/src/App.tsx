import React from "react";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import { Navigation } from "./components/navigation/Navigation";
import { Accessories } from "./pages/accessories/Accessories";
import { Home } from "./pages/Home";
import { Jackets } from "./pages/jackets/Jackets";
import { Shirts } from "./pages/shirts/shirts";

export const App = (): JSX.Element => {
  return (
    <>
      <Router>
        <Navigation />
        <Switch>
          <Route path="/jackets">
            <Jackets />
          </Route>
          <Route path="/shirts">
            <Shirts />
          </Route>
          <Route path="/accessories">
            <Accessories />
          </Route>
          <Route path="/">
            <Home />
          </Route>
        </Switch>
      </Router>
    </>
  );
};
