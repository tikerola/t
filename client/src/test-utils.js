import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { lightTheme } from "./styles/theme";
import { Router } from "react-router-dom";

import { createMemoryHistory } from "history";

const history = createMemoryHistory();
const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={lightTheme}>
      <Router history={history}>{children}</Router>
    </ThemeProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
