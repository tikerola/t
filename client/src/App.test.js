import React from "react";
import { App } from "./App";
import { render, fireEvent } from "./test-utils";

describe("<App />", () => {
  const resizeWindow = (width, height) => {
    global.innerWidth = width;
    global.innerHeight = height;
    global.dispatchEvent(new Event("resize"));
  };

  it("should match snapshot", () => {
    const component = render(<App />);

    expect(component.asFragment()).toMatchSnapshot();
  });

  it("should show burger, when width < theme.smallPad", () => {
    resizeWindow(500, 500);

    const component = render(<App />);

    const burger = component.queryByTestId("burger");
    expect(burger).toBeTruthy();
  });

  it("should open menu after pressing burger", async () => {
    const component = render(<App />);

    const burger = component.getByTestId("burger");
    fireEvent.click(burger);

    expect(component.queryByText(/home/i)).toBeTruthy();

    const accessories = component.getAllByText(/accessories/i);
    fireEvent.click(accessories[0]);

    expect(component.queryByText(/accessory paradise/i)).not.toBeTruthy();

    expect(component.queryByTestId(/spinner/i)).toBeTruthy();

    fireEvent.click(burger);
    const home = component.queryByText(/home/i);
    fireEvent.click(home);
    expect(component.queryByText(/accessory paradise/i)).toBeTruthy();

    fireEvent.click(burger);
    fireEvent.click(home);

    fireEvent.click(burger);
    const jackets = component.queryAllByText(/jackets/i);
    fireEvent.click(jackets[0]);
    expect(component.queryByText(/accessory paradise/i)).not.toBeTruthy();

    fireEvent.click(burger);
    fireEvent.click(home);

    fireEvent.click(burger);
    const shirts = component.queryAllByText(/shirts/i);
    fireEvent.click(shirts[0]);
    expect(component.queryByText(/accessory paradise/i)).not.toBeTruthy();
  });
});
