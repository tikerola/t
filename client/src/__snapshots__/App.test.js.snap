// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-hKgILt hGltGz"
  >
    <button
      class="sc-gsTCUz eBomCS"
      data-testid="burger"
    >
      <div />
      <div />
      <div />
    </button>
    <nav
      class="sc-dlfnbm xgSlL"
    >
      <a
        href="/"
      >
        Home
      </a>
      <a
        href="/jackets"
      >
        Jackets
      </a>
      <a
        href="/shirts"
      >
        Shirts
      </a>
      <a
        href="/accessories"
      >
        Accessories
      </a>
    </nav>
    <a
      aria-current="page"
      class="sc-eCssSg eRyLcw active"
      href="/"
    >
      <img
        alt="logo"
        src="globe.svg"
      />
    </a>
    <a
      class="sc-jSgupP zTMoV"
      href="/jackets"
    >
      Jackets
    </a>
    <a
      class="sc-jSgupP zTMoV"
      href="/shirts"
    >
      Shirts
    </a>
    <a
      class="sc-jSgupP zTMoV"
      href="/accessories"
    >
      Accessories
    </a>
  </div>
  <div
    class="sc-bdfBwQ keLBL"
  >
    <div>
      <h1
        class="sc-bBXqnf farLYQ"
      >
        Accessory Paradise
      </h1>
      <h1
        class="sc-iwyYcG juxsqh"
      >
        Hey there...
      </h1>
      <div
        class="sc-cxFLnm LIiWR"
      >
        <p
          class="sc-lmoMRL hYJjrK"
        >
          Browse through. We've got pretty much everything a person might ever need, like jackets, shirts, sunglasses, bracelets and shit...
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
